src/compas_fab/robots/targets.py

```diff
@@ -0,0 +1,449 @@
+        self.tolerance_position = tolerance_position
+        self.tool_coordinate_frame = tool_coordinate_frame
+
+    def __data__(self):
+        return {
+            "target_point": self.target_point,
+            "target_z_vector": self.target_z_vector,
+            "tolerance_position": self.tolerance_position,
+            "tool_coordinate_frame": self.tool_coordinate_frame,
+        }
+
+    def scaled(self, factor):
+        """Returns a copy of the target where the target point and tolerances are scaled.
+
+        Parameters
+        ----------
+        factor : float
+            The scaling factor.
+
+        Returns
+        -------
+        :class:`PointAxisTarget`
+            The scaled point-axis target.
+        """
+        target_point = self.target_point.scaled(factor)
+        tolerance_position = self.tolerance_position * factor if self.tolerance_position else None
+        target_z_vector = self.target_z_vector.scaled  # Vector is unitized and is not scaled
+        tool_coordinate_frame = self.tool_coordinate_frame.scaled(factor) if self.tool_coordinate_frame else None
+        return PointAxisTarget(target_point, target_z_vector, tool_coordinate_frame, tolerance_position, self.name)
+
+
+class ConfigurationTarget(Target):
+    """Represents a configuration target for the robot's end-effector motion planning.
+
+    The configuration target is a joint configuration of the robot's joints.
+    Given a ConfigurationTarget, the planner aims to find a path moving
+    the robot's joint positions to match with `ConfigurationTarget.target_configuration`.
+
+    ConfigurationTarget is suitable for targets whose joint configuration is known,
+    such as a home position, or a repetitive position that has been calibrated
+    in the actual robot cell, such as a tool changing position.
+
+    The number of joints in the target configuration should match the number of joints
+    in the robot's planning group. Otherwise the behaviour of the bankend planner may
+    be undefined. See tutorial :ref:`targets` for more details.
+
+    Attributes
+    ----------
+    target_configuration : :class:`compas_robots.Configuration`
+        The target configuration. joint_names and joint_values must be specified.
+        Defaults unit is radians for revolute and continuous joints, and meters for prismatic joints.
+    tolerance_above : :obj:`list` of :obj:`float`, optional
+        Acceptable deviation above the targeted configurations. One for each joint.
+        Always use positive values.
+        If not specified, the default value from the planner is used.
+    tolerance_below : :obj:`list` of :obj:`float`, optional
+        Acceptable deviation below the targeted configurations. One for each joint.
+        Always use positive values.
+        If not specified, the default value from the planner is used.
+    """
+
+    SUPPORTED_JOINT_TYPES = [Joint.PRISMATIC, Joint.REVOLUTE, Joint.CONTINUOUS]
```

Comment:
I'm not sure what ROSs tool supports Planar Joints, I have a feeling that the support for this type is really missing. 
https://answers.ros.org/question/59387/virtual-joint-as-part-of-planning-group-in-moveit/

> MoveIt! uses KDL for setting up the default kinematics solver. KDL does not support planar joints so that's why you are having trouble setting up a group like this. However, the fact that KDL does not support the planar joint only means that MoveIt! will not do the full inverse kinematics for you directly. You can still send individual joint values as a goal for the motion planning and MoveIt! will plan a path for you.

I think a PlanarJoint is more than 1 DOF. And I'm not aware of any planners that support this. 

https://robotics.stackexchange.com/questions/79027/moveit-group-arm-includes-joints-that-have-more-than-1-dof
> MoveIt itself doesn't really care, but plugins like KDL, LMA and others don't support joints with more than a single dof.

Here the Configuration Target relies on Configuration class, which I believe will not deal with a 2DOF value for one joint. So I think it is reasonable to exclude PLANAR Joint here. In fact, we should probably also add a note to the Configuration class to mention this problem.